go/interpreter/lfvm/instructions_test.go

```diff
@@ -1380,6 +1380,69 @@ func TestInstructions_EIP2929_SSTOREReportsOutOfGas(t *testing.T) {
 	}
 }
 
+func TestInstructions_StorageOps_CallStorageContext(t *testing.T) {
+	address := tosca.Address{}
+	_, _ = rand.Read(address[:])
+	key := tosca.Key{}
+	_, _ = rand.Read(key[:])
+	value := tosca.Word{}
+	_, _ = rand.Read(value[:])
+
+	tests := map[OpCode]struct {
+		implementation func(*context) error
+		stack          []uint256.Int
+	}{
+		SLOAD: {
+			implementation: opSload,
+			stack: []uint256.Int{
+				*new(uint256.Int).SetBytes(key[:]),
+			},
+		},
+		SSTORE: {
+			implementation: opSstore,
+			stack: []uint256.Int{
+				*new(uint256.Int).SetBytes(key[:]),
+				*new(uint256.Int).SetBytes(value[:]),
+			},
+		},
+	}
+
+	for op, test := range tests {
+		t.Run(op.String(), func(t *testing.T) {
+			forEachRevision(t, op, func(t *testing.T, revision tosca.Revision) {
+
+				ctxt := getEmptyContext()
+				ctxt.params.Recipient = address
+				ctxt.params.Revision = revision
+				ctxt.stack = fillStack(test.stack...)
+				ctxt.gas = tosca.Gas(math.MaxInt64)
+				runContext := tosca.NewMockRunContext(gomock.NewController(t))
+				if revision >= tosca.R09_Berlin {
+					runContext.EXPECT().AccessStorage(address, key).Return(tosca.WarmAccess)
+				}
+				if op == SLOAD {
+					runContext.EXPECT().GetStorage(address, key).Return(value)
+				}
+				if op == SSTORE {
+					runContext.EXPECT().SetStorage(address, key, value)
+				}
```

Comment:
adding `value` to stack could also be done here. Or this could be par of the test case parameters. It just seems a little unconsistent to me.

Reply:
I prefer to keep stack values in the declaration.
I agree that it is different from what we have been doing before; I find the callback with the mock context pretty verbose: it needs 3 lines in the declaration to declare a lambda, and requires still an if in case that it is left nil. 
This code is not like the other tests, but I find it more straight forward.